lib/request/index.js

```diff
@@ -0,0 +1,105 @@
+const _ = require('lodash'),
```

Comment:
Need to remove duplication with `lib/run/index.js` file.

We should break this and `run/index.js` file down into smaller functions with clear interfaces and update these files to use the functions.